src/index.ts

```diff
@@ -1,164 +1,634 @@
+import * as FissionAuthWnfsStaging from "./components/auth/implementation/fission-wnfs-staging.js"
+import * as FissionLobbyBase from "./components/confidences/implementation/fission-lobby.js"
+import * as FissionLobbyProduction from "./components/confidences/implementation/fission-lobby-production.js"
+import * as FissionLobbyStaging from "./components/confidences/implementation/fission-lobby-staging.js"
+import * as FissionReferenceProduction from "./components/reference/implementation/fission-production.js"
+import * as FissionReferenceStaging from "./components/reference/implementation/fission-staging.js"
+import * as ProperManners from "./components/manners/implementation/base.js"
+
+
+// RE-EXPORTS
+
+
+export * from "./components.js"
+export * from "./configuration.js"
+export * from "./common/types.js"
+export * from "./common/version.js"
+export * from "./permissions.js"
 
-import { AppInitOptions, InitialisationError, PermissionedAppInitOptions } from "./init/types.js"
-import { validateSecrets } from "./auth/state.js"
-import { bootstrapFileSystem, loadFileSystem } from "./filesystem.js"
+export * as did from "./did/index.js"
+export * as path from "./path/index.js"
+export * as ucan from "./ucan/index.js"
 
-import FileSystem from "./fs/index.js"
+export { AccountLinkingConsumer, AccountLinkingProducer } from "./linking/index.js"
+export { Confidences, FileSystemSecret } from "./confidences.js"
+export { FileSystem } from "./fs/filesystem.js"
+export { Session } from "./session.js"
 
-import { impl as auth } from "./auth/implementation.js"
-import { isPermissionedAppState } from "./auth/state.js"
-import { setImplementations } from "./setup.js"
-import { BASE_IMPLEMENTATION } from "./auth/implementation/base.js"
-import { USE_WNFS_IMPLEMENTATION } from "./auth/implementation/use-wnfs.js"
-import { LOBBY_IMPLEMENTATION } from "./auth/implementation/lobby.js"
-import { AppState } from "./auth/state/app.js"
-import { PermissionedAppState } from "./auth/state/permissionedApp.js"
-import * as appState from "./auth/state/app.js"
-import * as permissionedAppState from "./auth/state/permissionedApp.js"
+
+
+// TYPES & CONSTANTS
+
+
+export type Program = ShortHands & {
+  auth: AuthenticationStrategies
+  components: Components
+  confidences: {
+    collect: () => Promise<Maybe<string>> // returns username
+    request: () => Promise<void>
+    session: (username: string) => Promise<Maybe<Session>>
+  }
+  session: Maybe<Session>
+}
+
+
+export enum ProgramError {
+  InsecureContext = "INSECURE_CONTEXT",
+  UnsupportedBrowser = "UNSUPPORTED_BROWSER"
+}
+
+
+export type ShortHands = {
+  loadFileSystem: (username: string) => Promise<FileSystem>
+  loadRootFileSystem: (username: string) => Promise<FileSystem>
+}
+
+
+export type AuthenticationStrategies = Record<
+  string,
+  {
+    implementation: Auth.Implementation<Components>
+
+    accountConsumer: (username: string) => Promise<AccountLinkingConsumer>
```

Comment:
I wanted to separate the functions that a user/dev directly uses from the implementation. Mixing the implementation functions in here would be confusing, for example, `linkDevice` does not fully link a device, you have to create an account linking consumer & producer.